src/engine/bpmcontrol.cpp

```diff
@@ -874,14 +862,8 @@ void BpmControl::collectFeatures(GroupFeatureState* pGroupFeatures) const {
     if (getBeatContextNoLookup(dThisPosition,
                        dThisPrevBeat, dThisNextBeat,
                        &dThisBeatLength, &dThisBeatFraction)) {
-        pGroupFeatures->has_prev_beat = true;
-        pGroupFeatures->prev_beat = dThisPrevBeat;
-
-        pGroupFeatures->has_next_beat = true;
-        pGroupFeatures->next_beat = dThisNextBeat;
-
-        pGroupFeatures->has_beat_length = true;
-        pGroupFeatures->beat_length = dThisBeatLength;
+        pGroupFeatures->has_beat_length_sec = true;
+        pGroupFeatures->beat_length_sec = dThisBeatLength / m_pTrack->getSampleRate() / 2;
```

Comment:
This assumes a stereo signal. Does it assume that the file is stereo or that the signal passing through the engine is stereo?

Reply:
dThisBeatLength is the count of stereo samples. I will add a comment.